Add the initial Login Page

SillyRobot883 opened the pull request:
I added a new page for login with the initial design. 


![image](https://github.com/user-attachments/assets/99a6deb2-2502-4727-a480-13f4c81df9a8)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new login page with Arabic language support and right-to-left text direction.
	- Added styled buttons for Google and GitHub login options.
	- Enhanced visual presentation of the login interface with a new CSS file.
	- Improved footer layout and responsiveness with new styling.
	- Streamlined navigation component for better accessibility.
	- Introduced modern background styles with light and dark themes.

- **Style**
	- Implemented multiple style definitions for improved layout and design of the login page, footer, and background elements.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SillyRobot883 committed:
feat(frontend): add login page initial design

abo3skr2019 commented:
do we require first and last names to be accurate ? 
if we do we need to make the user input them manually because other wise it will be the name used with their google or github account
so i suggest creating a form for the user to give us the information needed 
the form will be a redirect from the oauth flow this redirect will be implemented by the backend team

abo3skr2019 commented:
ignore previous comments i already made it on my branch 
on PR i'm counting on the Frontend team to fix the css styling

abo3skr2019 commented:
you aren't on the latest dev 
and you aren't making use of the dynamic elements so it doesn't need to be a ejs file
and the buttons don't appear to be doing anything 
tie them into the Auth EndPoint

AFAskar reviewed (dismissed):
- Update to latest dev 
- Tie into Auth Endpoints
    - /backend/routes/auth.js
- And then Either
    - Stay on EJS and add some dynamic elements or partials 
    - or go back to a static html

SillyRobot883 committed:
feat(frontend): add google Oauth link to login.ejs

SillyRobot883 committed:
fix(frontend): fix header.ejs layout

SillyRobot883 committed:
refactor(frontend): update footer.ejs design

SillyRobot883 committed:
feat(frontend): add header and footer partials

SillyRobot883 committed:
fix(header.ejs): rename /register to /login

SillyRobot883 commented:
> * Update to latest dev
> * Tie into Auth Endpoints
>   
>   * /backend/routes/auth.js
> * And then Either
>   
>   * Stay on EJS and add some dynamic elements or partials
>   * or go back to a static html

I updated the branch to latest dev and added the required endpoints. Should be good now

abo3skr2019 commented on frontend/views/partials/footer.ejs:
```diff
@@ -1,23 +1,23 @@
-            <p class="text-muted">© 2025 وصل. جميع الحقوق محفوظة.</p>
+        <div class="row justify-content-center">
+            <div class="col-12 mb-4">
+                <h5 class="footer-brand">وصل</h5>
```
the button Looks Clickable so i suggest turning it to a href linking to /

abo3skr2019 commented on frontend/views/partials/header.ejs:
```diff
@@ -1,26 +1,22 @@
+            <ul class="navbar-nav me-auto mb-2 mb-lg-0">
                 <li class="nav-item">
-                    <a class="nav-link <%= currentPage === 'home' ? 'active' : '' %>" href="/">الرئيسة</a>
+                    <a class="nav-link <%= currentPage === 'home' ? 'active' : '' %>" aria-current="<%= currentPage === 'home' ? 'page' : 'false' %>" href="/">الرئيسة</a>
```
the "الرئيسة" button is redundant as the Logo act as it so you should remove it

abo3skr2019 commented on frontend/views/login.ejs:
the background of the floating login window is a bit empty but that is a bit of a nit pick

SillyRobot883 replied:
I added a background partial that can be used universally and added a background variant for later use

### To use it:
1.  import the necessary css file `<link href="/css/partials/background.css" rel="stylesheet">`
2. add the body to the background class `<body class="background">`

### Screenshots
light mode:
![image](https://github.com/user-attachments/assets/bb6f1406-fa13-4c1e-a7ff-95bfe0aa93f4)
dark mode:
![image](https://github.com/user-attachments/assets/5fd80415-56af-41e2-8a38-02bd2bb251bd)

AFAskar reviewed (changes requested):
the rest LGTM

SillyRobot883 committed:
refactor(header): make footer brand clickable

SillyRobot883 committed:
reactor(footer): remove footer brand underlining and change color to black

SillyRobot883 committed:
fix(header): remove redundant home page button

SillyRobot883 committed:
feat(frontned): add a background partial with dark mode option

SillyRobot883 committed:
remove redundant project name (وصل) from login message

AFAskar reviewed (approved):
Perfection
